query/iterator.gen.go

```diff
@@ -411,11 +413,57 @@ func (itr *floatSortedMergeIterator) pop() (*FloatPoint, error) {
 type floatSortedMergeHeap struct {
 	opt   IteratorOptions
 	items []*floatSortedMergeHeapItem
+	// if each input comes from a unique single time series, we can make a shortcut.
+	// detection of the shortcut introduces some overhead but it gets significant
+	// performance improvement in cases like SELECT * FROM m GROUP BY *
+	fast bool
+}
+
+func (h *floatSortedMergeHeap) detectFast() {
+	for _, item := range h.items {
+		if item.itr.Stats().SeriesN != 1 {
+			return
+		}
+	}
+
+	hasDup := false
+	s := make([]*floatSortedMergeHeapItem, len(h.items))
+	copy(s, h.items)
+
+	less := func(i, j int) bool {
```

Comment:
please investigate if something like the following would work as a comparator:

```
less := func(i, j int) bool {
  x, y := s[i].point, s[j].point
  hasDup = hasDup || x.Name == y.Name
  return ((x.Name < y.Name) || (x.Tags.ID() < y.Tags.ID()) && h.opt.Ascending
}
```
i think that should be at least as fast.

Reply:
@ayang64 the `less` function is required to sort ascending or descending order depending on `h.opt.Ascending`. It is unclear to me if the short version above achieves that. Also, it is setting `hasDup` to `true` only if `x.Name == y.Name`, whereas the original function sets `hasDup` to true iif both the `Name` and `Tags` are equal, which is a required property.

I didn't see any obvious performance issues with the existing code, and the overall improvement is significant.